### ISSUE TITLE
[Enhancement] Kafka Connect volume mounts

### ISSUE THREAD

**Issue by aka-peter** [strimzi/strimzi-kafka-operator#2571]
Is it possible to mount additional volumes into the Kafka Connect pods except for ConfigMaps or Secrets?
Our Use Case is to write event data to parquet files for later analysis. The FileStreamSinkConnector is included in the Strimzi container image so I am maybe missing something.
I found a similar question https://github.com/strimzi/strimzi-kafka-operator/issues/2445 but I fail to understand why scaling would be a problem when using a PVC. The same volume would be mounted in the different pods but the sink connectors would be responsible for not overwriting any data using partitioning.

**Comment by scholzj** [comment]
Actually, in many (most?) cases the volumes can be mounted only once (e.g. the AWS EBS volumes are read-write-once). Volumes which can be mounted for writing into multiple pods are really dependent on the environment. You are free to use the container image, and build your own deployment for this if you want.

**Comment by aka-peter** [comment]
Ok, thanks for the clarification. We are using MaprFS which is supporting RWX among several [others](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes).
We are currently using the increasingly not so open source Confluent Platform and are trying to get away from that. We probably then try to keep our Connect Clusters as is and use Strimzi  for Kafka if possible. It would be a very convenient feature in Strimzi for RWX volumes though.

**Comment by scholzj** [comment]
I added this to the tomorrow's Strimzi Community meeting agenda. We can discuss whether this is something we might consider in the future in which case we can change this to enhancement or whether we do not plan to support this in which case we can close it.

The meeting is tomorrow (July 30th) at 8:00 AM UTC https://zoom.us/j/93690955902 ... @aka-peter if you would be interested in joining but cannot do it tomorrow (because it is too short notice or bad timezone), we can also postpone it for August 13th at 4:00 PM UTC https://zoom.us/j/93355943577 ... please let us know.

**Comment by aka-peter** [comment]
I saw this just now. On vacation so I am not able to attend. We would use this feature for all our different file sink connetors and it would be very welcomed if it was added as an enhancement!

**Comment by scholzj** [comment]
Sorry, I didn't got to update it yet ... 

We decided on the Strimzi Community meeting to keep this open as enhancement. However - for the record - nobody is working on this right now. So if anyone wants to contribute this, feel free to start ... probably best by opening a [proposal](https://github.com/strimzi/proposals).

**Comment by cstmgl** [comment]
We are trying to do something similar (getting out of the confluent connectors) to use more generic/open source approaches but I'm also having a problem of not finding a correct way to add additional jars and libraries to the path.
At the moment I end up having to wrap the image in a docker but I was trying to not have to build my own image, and just provide the jars in the mount path, any other idea?

**Comment by scholzj** [comment]
Building a new image with the connectors you need is the right way to go. Using the connectors from some mounted volume goes against the most container best practices. There is an open proposal for having the image with the connectors build directly by Strimzi: https://github.com/strimzi/proposals/pull/21 ... feel free to have a look at it and comment there. the idea of this issue is about having volumes which the connectors can use to read data from / write data to. Not really for adding connectors.

**Comment by rao2100** [comment]
We have done the changes for this using using a fork of Strimzi 0.19
Our use case was for shared volume mounts, to enable file sink connector tasks to resume if they are  started on different node. 
Tested to be  working fine  with RWX on ceph-block storage and NFS.
Will forward port to strimzi master and do a pull request soon.

**Comment by scholzj** [comment]
Triaged on 17.3.2022: This is an enhancement which comes up often. Proposal will be needed to clarify how the storage will be handled (RWO versus RWX etc.)

**Comment by scholzj** [comment]
Triaged on 31.3.2022: We should close this and keep #3693 since that solves this problem in a most generic way.
